src/http/routes.cc

```diff
@@ -82,7 +82,12 @@ std::unique_ptr<reply> routes::exception_reply(std::exception_ptr eptr) {
         }
         std::rethrow_exception(eptr);
     } catch (const base_exception& e) {
-        rep->set_status(e.status(), json_exception(e).to_json());
+        if (e.content_type().size()) {
+            rep->set_status(e.status(), e.str());
```

Comment:
why aren't both conditional cases the same w.r.t. `set_status` (ie: `json_exception(e).to_json()`)? I was thinking the only thing happening was that the content type was being plumbed through for context.

Reply:
`json_exception` wraps the string up in a message like {"error": <e>, "status": 400}

When content_type is set, the caller is indicating that they want to send the whole request body themselves (i.e. e.str() is the whole body, not a string to go within a json object)

Reply:
ahh got it thanks